src/client/cli/cmd/set.cpp

```diff
@@ -124,16 +127,30 @@ mp::ParseCode cmd::Set::parse_args(mp::ArgParser* parser)
         else
         {
             const auto keyval = args.at(0).split('=', QString::KeepEmptyParts);
-            if (keyval.size() != 2 || keyval[0].isEmpty())
+            if ((keyval.size() != 1 && keyval.size() != 2) || keyval[0].isEmpty())
             {
                 cerr << "Bad key-value format.\n";
                 status = ParseCode::CommandLineError;
             }
-            else
+            else if (keyval.size() == 2)
             {
                 key = keyval.at(0);
                 val = keyval.at(1);
             }
+            else
+            {
+                mp::PlainPrompter prompter(term);
+                key = keyval.at(0);
+                try
+                {
+                    val = QString::fromStdString(prompter.prompt(key.toStdString()));
+                }
+                catch (const mp::ValueException& e)
+                {
+                    cerr << e.what() << std::endl;
+                    status = ParseCode::CommandLineError;
+                }
```

Comment:
Extract into a `checked_prompt` or something? `status = checked_prompt(prompt, key);`

Reply:
I'd have to pass `cerr` in, too (unless it was a private member of `Set`)… Since this is only used once, does it really help?

Reply:
Right, private. Reuse is not the only reason for small functional units. I think it would be better for readability, but I recognize it's not a big deal here.

I once watched a talk that really clicked with me and I really wish I still had the link to. The presenter asked (rephrasing from memory, but you get the idea): 

*When a friend asks you to describe your weekend, do you say "I got up, had breakfast, took a shower, chose my clothes, donned those clothes, chose my shoes, put them on, opened the door, exited the house, closed the door, inserted the key, turned the key 720° clockwise, went downstairs, entered the car..."? Or do you say "I went on a road trip, it was great", with perhaps a few other relevant details? It's the same with programming.*

It stuck with me. Not sure if I shared this yet. That's not a fair representation of the code above, but things tend to become more and more like that unless split out. I think `val = prompt(key)` would be more immediate for the reader, plus *code is read much more often than written*. But again, not a big deal here, just sharing where I am coming from.

Reply:
Done